src/main/java/sg/ncl/MainController.java

```diff
@@ -1918,6 +1948,94 @@ public String validateExperiment(
+        // extract just the particular node details to display
+        for (Map.Entry<String, Map<String, String>> nodesInfo : realization.getNodesInfoMap().entrySet()) {
+            String nodeName = nodesInfo.getKey();
+            Map<String, String> singleNodeDetailsMap = nodesInfo.getValue();
+            if (singleNodeDetailsMap.get("nodeId").equals(nodeId)) {
+                singleNodeInfoMap.put(nodeName, singleNodeDetailsMap);
+                // store the current os of the node into the form also
+                // have to pass the the services
+                saveImageForm.setCurrentOS(singleNodeDetailsMap.get("os"));
+            }
+        }
+
+        saveImageForm.setTeamId(teamId);
+        saveImageForm.setNodeId(nodeId);
+
+        model.addAttribute("teamName", teamName);
+        model.addAttribute("singleNodeInfoMap", singleNodeInfoMap);
+        model.addAttribute("pathTeamId", teamId);
+        model.addAttribute("pathExperimentId", expId);
+        model.addAttribute("pathNodeId", nodeId);
+        model.addAttribute("experimentName", realization.getExperimentName());
+        model.addAttribute("saveImageForm", saveImageForm);
+        return "save_experiment_image";
+    }
+
+    // bindingResult is required in the method signature to perform the JSR303 validation for Image object
+    @RequestMapping(value = "/experiments/save_image/{teamId}/{expId}/{nodeId}", method = RequestMethod.POST)
+    public String saveExperimentImage(
+            @Valid @ModelAttribute("saveImageForm") Image saveImageForm,
+            BindingResult bindingResult,
+            RedirectAttributes redirectAttributes,
+            @PathVariable String teamId,
+            @PathVariable String expId,
+            @PathVariable String nodeId) throws WebServiceRuntimeException, JsonProcessingException {
+
+        if (saveImageForm.getImageName().length() < 2) {
+            log.info("Save image form has errors {}", saveImageForm);
+            redirectAttributes.addFlashAttribute("message", "Image Name minimum 2 characters");
+            return "redirect:/experiments/save_image/" + teamId + "/" + expId + "/"  + nodeId;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        HttpEntity<String> request = createHttpEntityWithBody(mapper.writeValueAsString(saveImageForm));
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.saveImage(), HttpMethod.POST, request, String.class);
+
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                log.warn("Save image exception: {}", exceptionState);
+
+                switch (exceptionState) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)